### PR TITLE
Support `gitlab` as a federated auth source for `azd auth login`

### LINKED ISSUE #3242: Federated Credential Provider - What values are accepted?
**Output from `azd version`**
azd version 1.5.1 (commit 3856d1e98281683b8d112e222c0a7c7b3e148e96)

**Describe the issue**
I would like to use federated credentials in order to not use secrets in my GitLab pipeline to deploy my .Net Aspire orchesatrated app automatically to Azure Container Apps. When trying to sign in with the federated credential using `azd auth login --tenant-id "${AZURE_TENANT_ID}" --client-id "${AZURE_UAMI_CLIENT_ID}" --federated-credential-provider "${FED_CRED_PROVIDER_ID}" --no-prompt` where the client-id is the ID of the application registration with the federated credential and the provider being `api://AzureADTokenExchange`, I get the following error: 

ERROR: logging in: unsupported federated token provider: 'api://AzureADTokenExchange'

**To Reproduce**
Create a federated credential for GitLab by following this guide: https://techcommunity.microsoft.com/t5/fasttrack-for-azure/azure-access-from-github-and-gitlab-pipelines-without-secrets/ba-p/3858885

Execute the above mentioned command and pass the values accordingly, though it does not matter what value is used for the client-id. The error will appear. From my testing, only "github" was accepted, which is not feasible for me.

### PR DESCRIPTION
**Pull request by ellismg**
Unlike GitHub, instead of having to fetch a token via an HTTP endpoint, you can just ask GitLab to inject the token into an env-var. We expect that the user will add the following to their pipeline:

```
id_tokens:
  AZD_GITLAB_FEDERATED_TOKEN:
    aud: "api://AzureADTokenExchange"
```

Then, `azd auth login --federated-credential-provider gitlab` should work as expected.

Fixes #3242

### CODE REVIEW

**Comment by ellismg** [comment]
@MiTschMR, see if this works for you.

When I first did the federated credential feature, I think we supported just passing the raw value of the federated token using `--federated-token`.  That didn't work great for GitHub because of it's token refresh scheme, but maybe it's a good idea to have that as an escape hach.  If it had existed, @MiTschMR could have used that instead of needing us to add explicit support for GitLab.

File: cli/azd/pkg/auth/manager.go [hunk 1]
```diff
@@ -422,12 +420,26 @@ func (m *Manager) newCredentialFromFederatedTokenProvider(
...
+					return "", fmt.Errorf("fetching federated token: %w", err)
+				}
+
+				return federatedToken, nil
+			case gitLabFederatedAuth:
+				federatedToken, has := os.LookupEnv("AZD_GITLAB_FEDERATED_TOKEN")
+				if !has {
+					return "",
+						errors.New("AZD_GITLAB_FEDERATED_TOKEN is not set. It should be a valid GitLab id_token with an " +
+							" an audience of 'api://AzureADTokenExchange'")
```
Comment (weikanglim) [inline cli/azd/pkg/auth/manager.go:436]:
```suggestion
						errors.New("AZD_GITLAB_FEDERATED_TOKEN is not set. It should be a valid GitLab id_token with an " +
							" audience of 'api://AzureADTokenExchange'")
```

File: cli/azd/pkg/auth/manager.go [hunk 2]
```diff
@@ -422,12 +420,26 @@ func (m *Manager) newCredentialFromFederatedTokenProvider(
...
-			}
+			switch provider {
+			case gitHubFederatedAuth:
+				federatedToken, err := m.ghClient.TokenForAudience(ctx, "api://AzureADTokenExchange")
+				if err != nil {
+					return "", fmt.Errorf("fetching federated token: %w", err)
+				}
+
+				return federatedToken, nil
+			case gitLabFederatedAuth:
```
Comment (weikanglim) [inline cli/azd/pkg/auth/manager.go:431]: I was really curious about what the "standard" is and looked around a little.
I think we can drop all the custom support and instead implement OIDC token support generically, following the footsteps of what's been done in other tools ([Pulumi](https://www.pulumi.com/blog/oidc-with-azure/#enabling-oidc), [Terraform](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/service_principal_oidc#configuring-the-service-principal-in-terraform)) for better compatibility:

- `ACTIONS_ID_TOKEN_REQUEST_TOKEN` and `ACTIONS_ID_TOKEN_REQUEST_URL` are automatically detected for GitHub
- `ARM_OIDC_REQUEST_TOKEN` and `ARM_OIDC_REQUEST_TOKEN_URL` for override
- `ARM_OIDC_TOKEN` for direct token

Be happy to create an issue / make the changes here if you think it's worthwhile.

Comment (ellismg) [reply cli/azd/pkg/auth/manager.go:431]: Love it - feel free to take over the PR.

**Pull request closed without merging**